Initalize git repo when creatin new rails app

dixpac opened the pull request:
Since Rails is already generating `.gitignore`, and in 90% of cases when creating new rails app user initialise git repo. Now Rails automatically initialises new git repo on `rails new` command  If `--skip-git` is not specified.

I've added this as new "step" in generator called `version_control` in case someone wants to override this when creating custom app builder (maybe not using git as version_control but something else)

dixpac committed:
Initalize git repo when creatin new rails app

* If `--skip-git` is not specified initalize git repo
 when creating new rails app

zachweed commented:
There's a large difference between creating a single, small file, and initializing an empty git repository automatically. Also, where is the source indicating that 90% of rails applications use git?

That aside.. If this does get approved then I would highly recommend that we flip the `--skip-git` to a flag like `--with-git` so that we assume nothing about what the engineer might be using for version control and allow them to opt-in, if they so choose. In addition to that, it would make sense if you were to also include changes to only create `.gitignore` when a flag is specified (or not specified).

dixpac commented:
@zachweed thanks for the comment :)
 Sure there is the huge difference(56k to be exact :)), but since rails generator is already generating `.gitignore` that means that assumption is made you are going to use git. If that is the case why not to initialize the git repo ?

That being said, I like your idea about using `--with-git` instead of `--skip-git` (that was also my initial idea), but since there is already `--skip-git`, I push this solution to see the reactions about adding this kind of feature to the generator :)

zachweed commented:
@dixpac I completely agree. I personally don't agree with the `.gitignore` being created either, which is why I was _slightly_ implying removal of that as well by using something with  `--with-git`. Feel free to ping me if you need any help with any of it :)

georgeclaghorn commented:
“Assume nothing“ isn’t really Rails’s [modus operandi](http://rubyonrails.org/doctrine/#omakase).

rafaelfranca commented:
Thank you for the pull request but doing this by default means that users now need to have git installed to generated a rails application otherwise it will fail. Can we check if git is present in the machine before running the command?

dixpac commented:
@rafaelfranca  we can check if git is present, but now sure do we have to because `run` command will return `false` if git is not present and nothing will happen (execution will not fail), new rails app will be initialized normally(without git)

zachweed commented:
@dixpac it may be a good idea to return a message indicating we intended to create an empty git repo but were unable to due to it not being installed.

PritiKumr commented on railties/lib/rails/generators/rails/app/app_generator.rb:
```diff
@@ -205,6 +211,7 @@ def create_root_files
         build(:configru)
         build(:gitignore)   unless options[:skip_git]
         build(:gemfile)     unless options[:skip_gemfile]
+        build(:version_control)
```
Why not check the `options[:skip_git]` flag here like how its been done for `:gitignore`?

dixpac replied:
Since other developers can create custom AppBuilder templates by overriding rails AppBuilder methods, adding `options[:skip_git]` here would add more complexity if someone wants to use other `version_control` system in their templates :)

rafaelfranca reviewed (changes requested):
Could you add a CHANGELOG entry?

dixpac commented:
@rafaelfranca  sure I added [CHANGELOG](https://github.com/rails/rails/pull/27632/commits/d65680cb05197597533ff5afee14910f08a69936#diff-1846d4882fa63eb5b1271541995cbfdcR1) and rebased from master :)

dixpac commented:
@rafaelfranca  I'm not sure why tests are red after I rebased from master :(

maclover7 commented:
@dixpac restarted the randomly failing build, all green now

welkie commented:
I actually like this idea. At first I was thinking "wow, that assumes a lot about the developer's intentions" but then I realized... Yes, and that's probably a good thing in the context of Rails development. Rails is opinionated. Most developers use version control and most developers using version control use Git for it. So I think it is logical to implicitly create a new Git repo during the creation of a new Rails app.
